bundles/org.eclipse.mita.platform/src/org/eclipse/mita/platform/scoping/PlatformDslResourceDescriptionStrategy.java

```diff
@@ -47,6 +60,11 @@ public boolean createEObjectDescriptions(EObject eObject, IAcceptor<IEObjectDesc
 			return false;
 		} else if(eObject instanceof Platform) {
 			return createPlatformDescription((Platform) eObject, acceptor);
+		} else if( (eObject instanceof StructureType) 
+				|| (eObject instanceof SumAlternative) 
+				|| (eObject.eContainer() instanceof SumAlternative && !(eObject instanceof TypeSpecifier))) {
+			export(eObject, acceptor);
```

Comment:
Is there a reason why we ignore the "exported" flag here and define that **all** structs and sum types are exported here?

Reply:
A more generic solution would be to create EObjectDescriptions for every element that is of type `Exportable` and where `isExported==true`. This would also include enums, native types etc.

Reply:
In platform description one needs to set the `export` flag. This is additional effort, but also allows to specify which types shall be visible and which not.

Reply:
It seems to me that it doesn't make sense to not export types from your platform since you can't use them anyway (see #6). Also declaring types inside the platform that aren't exported can't be used in the platform, too, since the platform doesn't contain any programming logic. In contrast, types that are used in setups should always be visible.

I exported SumTypes and StructureTypes explicitly since I think those are the only ones that need exporting. I'm not sure if GeneratedTypes should be exported, nor if a platform should be able to define PrimitiveTypes.